src/test/java/org/opensearch/knn/integ/FilteredSearchBinaryIT.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.integ;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.SneakyThrows;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.opensearch.client.Response;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.knn.KNNMappingsBuilder;
+import org.opensearch.knn.index.KNNSettings;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.index.query.KNNQueryBuilder;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.util.List;
+import java.util.Locale;
+
+import static org.opensearch.knn.common.KNNConstants.METHOD_HNSW;
+
+public class FilteredSearchBinaryIT extends AutoIndexCleanUpTestCase {
```

Comment:
I saw there also have a new test class `BinaryIndexIT`, can we merge binary related ITs into one test file?

Reply:
I think multiple smaller classes are better than single big class.